### PR TITLE
A few clean ups for Au type

### PR DESCRIPTION
**Pull request by glennw**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7766)

### CODE REVIEW

**Comment by Ms2ger** [comment]
None of those changes feel like improvements to me. Can you explain why you made them?

**Comment by glennw** [comment]
Removal of the unused snapping related methods is to discourage pixel snapping during layout, so that sub-pixel accuracy remains until rendering. The pt related functions are moved because they rely on hard coded assumptions and are used only by the mac font code. The zero point change is to be consistent with other code (everywhere else uses ZERO_POINT).

(I should have been more explicit with the original commit message).

File: components/gfx/platform/macos/font.rs [hunk 1]
```diff
@@ -161,27 +175,27 @@ impl FontHandleMethods for FontHandle {
         let scale = px_to_pt(self.ctfont.pt_size() as f64) / (ascent + descent);
         let line_gap = (ascent + descent + leading + 0.5).floor();
 
-        let max_advance_width = Au::from_pt(bounding_rect.size.width as f64);
+        let max_advance_width = au_from_pt(bounding_rect.size.width as f64);
```
Comment (pcwalton) [inline components/gfx/platform/macos/font.rs]: Why this change?
